testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeDatabricksInsertCompatibility.java

```diff
@@ -373,6 +373,32 @@ public void verifyCompressionCodecsDataProvider()
                                         .collect(toImmutableList())));
     }
 
+    @Test(groups = {DELTA_LAKE_DATABRICKS, PROFILE_SPECIFIC_TESTS})
+    public void testPreventingWritesToTableWithNotNullableColumns()
+    {
+        String tableName = "test_preventing_inserts_into_table_with_not_nullable_columns_" + randomTableSuffix();
+
+        try {
+            onDelta().executeQuery("CREATE TABLE default." + tableName + "( " +
+                    " id INT NOT NULL, " +
+                    " a_number INT) " +
+                    "USING DELTA " +
+                    "LOCATION 's3://" + bucketName + "/databricks-compatibility-test-" + tableName + "'");
+
+            onDelta().executeQuery("INSERT INTO " + tableName + " VALUES(1,1)");
+            assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO " + tableName + " VALUES (2, 2)"))
+                    .hasMessageContaining("Inserts are not supported for tables with non-nullable columns");
```

Comment:
Can we add another assertion to verify that the data wasn't inserted? I know it's ensured in the bottom assertion though.

Reply:
I am not sure I understand your idea. Do you want me to add another
```
assertThat(onTrino().executeQuery("SELECT * FROM " + tableName))
                    .containsOnly(row(1, 2));
```
after 
```
assertQueryFailure(() -> onTrino().executeQuery("INSERT INTO " + tableName + " VALUES (2, 2)"))
                    .hasMessageContaining("Inserts are not supported for tables with non-nullable columns");
```
?

Reply:
Right, I think it should be `row(1, 1)`, not `row(1, 2)`.

Reply:
oh right, I get it, thanks !